ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4

```diff
@@ -663,11 +664,11 @@ DECIMAL_VALUE
     ;
 
 IDENTIFIER
-    : (LETTER | '_') (LETTER | DIGIT | '_' | '@' | ':')*
+    : (LETTER | '_') (LETTER | DIGIT | '_' | '@' )*
```

Comment:
AS you say, it probably wouldn't work as its an invalid java identifier. Such a break is OK though. We just need to clearly document it.

Reply:
This won't affect the backward compatibility. As you mentioned `:` is not permitted in the java identifier and regardless we cannot use it in identifier name.
I think we should remove `@` for the same reason but this can be done in a separate PR.